tests/lib/cli.js

```diff
@@ -695,13 +701,39 @@ describe("cli", () => {
                 describe("when given a pattern to ignore", () => {
                     it(`should not process any files with configType:${configType}`, async () => {
                         const ignoredFile = getFixturePath("cli/syntax-error.js");
+                        const ignorePathOption = useFlatConfig
+                            ? ""
+                            : "--ignore-path .eslintignore_empty";
                         const filePath = getFixturePath("cli/passing.js");
-                        const exit = await cli.execute(`--ignore-pattern cli/ ${ignoredFile} ${filePath}`, null, useFlatConfig);
+                        const ignorePattern = useFlatConfig ? "cli/**" : "cli/";
+                        const exit = await cli.execute(
+                            `--ignore-pattern ${ignorePattern} ${ignorePathOption} ${ignoredFile} ${filePath}`, null, useFlatConfig
+                        );
```

Comment:
This test was not working as intended because all the files were already ignored by [`.eslintignore`](https://github.com/eslint/eslint/blob/main/tests/fixtures/.eslintignore), so `--ignore-pattern` actually had no effect.